File: archive/v2-csharp-avalonia/src/Spanova.Core/Model/Girder.cs
namespace Spanova.Core.Model;

/// <summary>
/// Superstructure girder sizing for one bridge alternative (spec section
/// 9). Prototype scope is the first bridge type only - PRECAST GIRDER
/// VIADUCT (spec section 8) - so one uniform girder type per alternative.
/// </summary>
public sealed class Girder
{
    public int Count { get; init; }

    /// <summary>Girder depth, in meters.</summary>
    public double DepthM { get; init; }

    /// <summary>Girder spacing (centre-to-centre), in meters.</summary>
    public double SpacingM { get; init; }
}

File: backend/bridge-core/src/main/java/com/spanova/bridgecore/Girder.java
package com.spanova.bridgecore;

/**
 * Superstructure girder sizing for one bridge alternative (spec section
 * 9). Prototype scope is the first bridge type only - PRECAST GIRDER
 * VIADUCT (spec section 8) - so one uniform girder type per alternative.
 */
public record Girder(int count, double depthM, double spacingM) {
}
